mirdata/datasets/saraga_carnatic.py

```diff
@@ -152,57 +149,29 @@ def __init__(self, track_id, data_home):
             [self._data_home, self._track_paths['metadata'][0]]
         )
 
-        # Flag to separate between carnatinc and hindustani tracks
-        self.iam_style = str(self.track_id.split('_')[0])
-
-        # CARNATIC MUSIC TRACKS
-        if self.iam_style == 'carnatic':
-            metadata = DATA.metadata(self.metadata_path)
-            if metadata is not None and track_id == metadata['track_id']:
-                self._track_metadata = metadata
-            else:
-                # annotations with missing metadata
-                self._track_metadata = {
-                    'raaga': None,
-                    'form': None,
-                    'title': None,
-                    'work': None,
-                    'length': None,
-                    'taala': None,
-                    'album_artists': None,
-                    'mbid': None,
-                    'artists': None,
-                    'concert': None,
-                }
-
-        # HINDUSTANI MUSIC TRACKS
-        if self.iam_style == 'hindustani':
-            metadata = DATA.metadata(self.metadata_path)
-            if metadata is not None and track_id == metadata['track_id']:
-                self._track_metadata = metadata
-            else:
-                # annotations with missing metadata
-                self._track_metadata = {
-                    'title': None,
-                    'raags': None,
-                    'length': None,
-                    'album_artists': None,
-                    'forms': None,
-                    'mbid': None,
-                    'artists': None,
-                    'release': None,
-                    'works': None,
-                    'taals': None,
-                    'layas': None,
-                }
-
-        # Common attributes for Hindustani and Carnatic tracks
+        # Track attributes
+        metadata = DATA.metadata(self.metadata_path)
+        if metadata is not None and metadata['title'].replace(' ', '_') in self.track_id:
+            self._track_metadata = metadata
+        else:
+            # annotations with missing metadata
```

Comment:
do we have many files with missing metadata? maybe we can correct them or talk to the people who created the dataset to see what happened.

Reply:
In fact, there are no files with missing metadata. I wrote that just in case. Should I remove it?

Reply:
I think so, yes. then we will catch any errors in missing metadata and try to correct it.

Reply:
Cool! I'll do that before running the `test_full_dataset` in both loaders.

Reply:
Done. All the tracks have valid metadata!!

Reply:
Well in order to pass the tests I need to leave this missing metadata conditional. In fact, it is present in every loader. Maybe it can be useful if a metadata file is missing for some reason!